runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java

```diff
@@ -99,48 +104,119 @@ else if (event.getKey().equals("showEssence"))
 	{
-		if (event.getGameState().equals(GameState.LOGGED_IN))
+		if (config.showHerbBoxes() && !hasSentHerbMsg && checkCanCollectHerbBox())
 		{
-			if (config.showHerbBoxes() && !hasSentHerbMsg && checkCanCollectHerbBox())
-			{
-				sendChatMessage("You have herb boxes waiting to be collected at NMZ.");
-				hasSentHerbMsg = true;
-			}
-			if (config.showStaves() && !hasSentStavesMsg && checkCanCollectStaves())
-			{
-				sendChatMessage("You have staves waiting to be collected from Zaff.");
-				hasSentStavesMsg = true;
-			}
-			if (config.showEssence() && !hasSentEssenceMsg && checkCanCollectEssence())
-			{
-				sendChatMessage("You have pure essence waiting to be collected from Wizard Cromperty.");
-				hasSentEssenceMsg = true;
-			}
+			sendChatMessage("You have herb boxes waiting to be collected at NMZ.");
+			hasSentHerbMsg = true;
+		}
+
+		if (config.showStaves() && !hasSentStavesMsg && checkCanCollectStaves())
+		{
+			sendChatMessage("You have staves waiting to be collected from Zaff.");
+			hasSentStavesMsg = true;
+		}
+
+		if (config.showEssence() && !hasSentEssenceMsg && checkCanCollectEssence())
+		{
+			sendChatMessage("You have pure essence waiting to be collected from Wizard Cromperty.");
+			hasSentEssenceMsg = true;
+		}
+
+		if (config.showSand() && !hasSentSandMsg && checkCanCollectSand())
+		{
+			sendChatMessage("You have buckets of sand waiting to be collected from Bert.");
+			hasSentSandMsg = true;
+		}
+
+		if (config.showEcto() && !hasSentEctoMsg && checkCanCollectEcto())
+		{
+			sendChatMessage("You have bonemeal and buckets of slime waiting to be collected from Robin.");
+			hasSentEctoMsg = true;
 		}
 	}
 
 	private boolean checkCanCollectHerbBox()
 	{
 		int value = client.getSetting(Varbits.DAILY_HERB_BOX);
-		return value < 15; // < 15 can claim
+		return prevHerbVarbVal == value ? (prevHerbVarbVal = value) < 15 : false;
 	}
 
 	private boolean checkCanCollectStaves()
 	{
-		int value = client.getSetting(Varbits.DAILY_STAVES);
-		return value == 0; // 1 = can't claim
+		if (didCompleteDiaries(Varbits.DIARY_VARROCK_EASY, Varbits.DIARY_VARROCK_MEDIUM, Varbits.DIARY_VARROCK_HARD, Varbits.DIARY_VARROCK_ELITE))
+		{
+			int value = client.getSetting(Varbits.DAILY_STAVES);
+			return prevStavesVarbVal == value ? (prevStavesVarbVal = value) == 0 : false;
+		}
+		return false;
 	}
 
 	private boolean checkCanCollectEssence()
 	{
-		int value = client.getSetting(Varbits.DAILY_ESSENCE);
-		return value == 0; // 1 = can't claim
+		if (didCompleteDiaries(Varbits.DIARY_ARDOUGNE_EASY, Varbits.DIARY_ARDOUGNE_MEDIUM, Varbits.DIARY_ARDOUGNE_HARD, Varbits.DIARY_ARDOUGNE_ELITE))
+		{
+			int value = client.getSetting(Varbits.DAILY_ESSENCE);
+			return prevEssVarbVal == value ? (prevEssVarbVal = value) == 0 : false;
+		}
+		return false;
+	}
+
+	private boolean checkCanCollectSand()
+	{
+		if (client.getSetting(Varbits.QUEST_A_HAND_IN_THE_SAND) == 160 && !FREE_TO_PLAY_WORLDS.contains(client.getWorld()))
```

Comment:
Completing the ardy elite diaries makes buckets of sand be automatically delivered to your bank, so this notification has no use at that point.

Reply:
Fair point, I'll remove this later tonight or tomorrow